File: music-android/app/src/main/java/com/sismics/music/ui/form/validator/Required.java
package com.sismics.music.ui.form.validator;

import android.content.Context;

import com.sismics.music.R;

/**
 * Text presence validator.
 * 
 * @author bgamard
 */
public class Required implements ValidatorType {

    @Override
    public boolean validate(String text) {
        return text.trim().length() != 0;
    }

    @Override
    public String getErrorMessage(Context context) {
        return context.getString(R.string.validate_error_required);
    }

}

File: music-android/app/src/main/java/com/sismics/music/util/form/validator/Required.kt
package com.sismics.music.util.form.validator

import android.content.Context

import com.sismics.music.R

/**
 * Text presence validator.

 * @author bgamard
 */
class Required : ValidatorType {

    override fun validate(text: String): Boolean {
        return text.trim { it <= ' ' }.isNotEmpty()
    }

    override fun getErrorMessage(context: Context): String {
        return context.getString(R.string.validate_error_required)
    }

}
